extensions-core/druid-lookups/src/main/java/io/druid/server/lookup/DataFetcher.java

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ *  Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ *  or more contributor license agreements. See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership. Metamarkets licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package io.druid.server.lookup;
+
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.druid.server.lookup.jdbc.JdbcDataFetcher;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class will be used to pull data from a given lookup table.
+ *
+ * @param <K> Keys type
+ * @param <V> Values type
+ */
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name = "jdbcDataFetcher", value = JdbcDataFetcher.class)
+})
+public interface DataFetcher<K, V>
+{
+  /**
+   * Function used to fetch all the pair of key-values from the lookup
+   * One use case of this method is to populate a polling cache {@link io.druid.server.lookup.cache.polling.PollingCache}
+   *
+   * @return Returns a map containing All the key-value pairs.
+   */
+  Map<K, V> fetchAll();
```

Comment:
@drcrallen i am not sure what is the difference between returning a map or a collection view of the map ?